flake.nix

```diff
@@ -0,0 +1,356 @@
+          # devLibPath = pkgs.lib.makeLibraryPath [ pkgs.libressl.dev ];
+          # pname = "libressl";
+          # version = "3.9.2";
+          # includes = [ "tests/tlstest.sh" ];
+          # src = pkgs.fetchurl {
+          #   url = "mirror://openbsd/LibreSSL/${pname}-${version}.tar.gz";
+          #   hash = "sha256-ewMdrGSlnrbuMwT3/7ddrTOrjJ0nnIR/ksifuEYGj5c=";
+          # };
+          # nativeBuildInputs = _old.nativeBuildInputs ++ (with pkgs; [
+          #   pkg-config
+          # ]);
+
+          # Patched according to the previous versions:
+          # https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/development/libraries/libressl/default.nix#L118
+          # For unknown reasons the newer versions are not patched this way (yet?)
+          patches = [
+            (pkgs.fetchpatch {
+              url = "https://github.com/libressl/portable/commit/86e4965d7f20c3a6afc41d95590c9f6abb4fe788.patch";
+              includes = [ "tests/tlstest.sh" ];
+              hash = "sha256-XmmKTvP6+QaWxyGFCX6/gDfME9GqBWSx4X8RH8QbDXA=";
+            })
+          ];
+
+          # NOTE: Due to name conflicts between OpenSSL and LibreSSL we need to resolve this manually.
+          postFixup =  pkgs.lib.concatLines [ 
+            ( _old.postFixup or "" )
+            ''
+            cp $dev/lib/pkgconfig/libcrypto.pc $dev/lib/pkgconfig/libresslcrypto.pc
+            sed --in-place --expression 's/-lcrypto/-lresslcrypto/' $dev/lib/pkgconfig/libresslcrypto.pc
+            ln -s $out/lib/libcrypto.so $out/lib/libresslcrypto.so
+            ln -s $out/lib/libcrypto.a $out/lib/libresslcrypto.a
+            ''
+          ];
+
+        });
+        # Current list of targets: tomcrypt botan cryptopp openssl boringssl gcrypt mbedtls ippcp nettle libressl
+        tomcryptShim = import ./nix/tomcryptshim.nix { inherit pkgs libtomcrypt libtommath; };
+        botanShim = import ./nix/botanshim.nix { inherit pkgs; };
+        cryptoppShim = import ./nix/cryptoppshim.nix { inherit pkgs cryptopp; };
+        opensslShimBuilder = { version, hash }: import ./nix/opensslshim.nix { inherit pkgs; openssl = (openssl { version = version; hash = hash;}); };
+        boringsslShim = import ./nix/boringsslshim.nix { inherit pkgs; boringssl = patched_boringssl; };
+        gcryptShim = import ./nix/gcryptshim.nix { inherit pkgs libgcrypt libgpg-error; };
+        mbedtlsShim = import ./nix/mbedtlsshim.nix { pkgs = pkgs; };
+        ippcpShim = import ./nix/ippcpshim.nix { pkgs = pkgs; ipp-crypto = customPkgs.ipp-crypto; };
+        nettleShim = import ./nix/nettleshim.nix { inherit pkgs nettle; };
+        libresslShim = import ./nix/libresslshim.nix { inherit pkgs libressl; };
+
+        overlays = [];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+        customPkgs = import custom-nixpkgs {
+          inherit system overlays;
+        };
```

Comment:
Maybe move these lines up? As they are used earlier would make sense (readability-wise) to have them at the top if possible.